client/lib/plugins/sanitize-section-content.js

```diff
@@ -0,0 +1,246 @@
+			}
+
+		default:
+			return false;
+	}
+};
+
+const isValidYoutubeEmbed = node => {
+	if ( node.nodeName.toLowerCase() !== 'iframe' ) {
+		return false;
+	}
+
+	if ( node.getAttribute( 'class' ) !== 'youtube-player' ) {
+		return false;
+	}
+
+	if ( node.getAttribute( 'type' ) !== 'text/html' ) {
+		return false;
+	}
+
+	const link = document.createElement( 'a' );
+	link.href = node.getAttribute( 'src' );
+
+	return (
+		validUrl.isWebUri( node.getAttribute( 'src' ) ) &&
+		( link.hostname === 'youtube.com' || link.hostname === 'www.youtube.com' )
+	);
+};
+
+const replacementFor = node => {
+	const tagName = node.nodeName.toLocaleLowerCase();
+
+	switch ( tagName ) {
+		case 'h1':
+		case 'h2':
+			return 'h3';
+
+		default:
+			return null;
+	}
+};
+
+/**
+ * Sanitizes input HTML for security and styling
+ *
+ * @param {String} content unverified HTML
+ * @returns {string} sanitized HTML
+ */
+export const sanitizeSectionContent = content => {
+	const parser = new DOMParser();
+	const doc = parser.parseFromString( content, 'text/html' );
+
+	// this will let us visit every single DOM node programmatically
+	const walker = doc.createTreeWalker( doc.body );
+
+	/**
+	 * we don't want to remove nodes while walking the tree
+	 * or we'll invite data-race bugs. instead, we'll track
+	 * which ones we want to remove then drop them at the end
+	 *
+	 * @type {Array<Node>} List of nodes to remove
+	 */
+	const removeList = [];
+
+	/**
+	 * track any tags we want to replace, we'll need to
+	 * transfer the children too when we do the swap
+	 *
+	 * @type {Array<Node, Node>} List of pairs of nodes and their replacements
+	 */
+	const replacements = [];
+
+	// walk over every DOM node
+	while ( walker.nextNode() ) {
+		const node = walker.currentNode;
+		const tagName = node.nodeName.toLowerCase();
+
+		if ( ! isAllowedTag( tagName ) && ! isValidYoutubeEmbed( node ) ) {
+			removeList.push( node );
+			continue;
+		}
+
+		const replacement = replacementFor( node );
+		if ( replacement ) {
+			replacements.push( [ node, document.createElement( replacement ) ] );
+		}
+
+		// strip out anything not explicitly allowed
+		// in the attributes. we want to eliminate
+		// potential cross-site scripting attacks _and_
+		// prevent custom styles from interfering with
+		// our page's own rendering
+
+		const attrRemoveList = [];
+
+		// these aren't Arrays, so iteration is odd
+		forEach( node.attributes, ( { name: attrName, value } ) => {
+			// get rid of non-whitelisted attributes
+			if ( ! isAllowedAttr( tagName, attrName ) ) {
+				attrRemoveList.push( attrName );
+				return;
+			}
+
+			// for some, specifically links, we don't want to
+			// "sanitize" the values because we could mess them up
+			// for example, encoding a real URL will break it
+			if (
+				( ( 'src' === attrName || 'href' === attrName ) && validUrl.isWebUri( value ) ) ||
+				( 'iframe' === tagName && 'type' === attrName && isValidYoutubeEmbed( node ) )
+			) {
+				return;
+			}
```

Comment:
Once we stopped encoding attribute values, this `if` statement no longer does anything useful -- or does it?

Reply:
Good catch! Removed in 0df9c6e42